diff --git a/src/dlio_profiler/brahma/posix.h b/src/dlio_profiler/brahma/posix.h
index 60a71de..6d03768 100644
--- a/src/dlio_profiler/brahma/posix.h
+++ b/src/dlio_profiler/brahma/posix.h
@@ -10,12 +10,12 @@
 #include <dlio_profiler/core/macro.h>
 #include <dlio_profiler/dlio_logger.h>
 #include <fcntl.h>
-#include <filesystem>
+#include <experimental/filesystem>
 #include <fstream>
 #include <sys/param.h>
 #include <vector>
 
-namespace fs = std::filesystem;
+namespace fs = std::experimental::filesystem;
 
 namespace brahma {
     class POSIXDLIOProfiler : public POSIX {
diff --git a/src/dlio_profiler/brahma/stdio.h b/src/dlio_profiler/brahma/stdio.h
index 34d5875..f8ad1f6 100644
--- a/src/dlio_profiler/brahma/stdio.h
+++ b/src/dlio_profiler/brahma/stdio.h
@@ -10,11 +10,11 @@
 #include <dlio_profiler/dlio_logger.h>
 #include <dlio_profiler/core/macro.h>
 #include <fcntl.h>
-#include <filesystem>
+#include <experimental/filesystem>
 #include <fstream>
 #include <vector>
 
-namespace fs = std::filesystem;
+namespace fs = std::experimental::filesystem;
 
 namespace brahma {
     class STDIODLIOProfiler : public STDIO {
